### PR TITLE
Better stream events fetch logic.

### PR DESCRIPTION
**Pull request by chtyim**
This change only affect the get stream events REST API.

The improvement is mainly for the case when a fetch request comes with
a startTime, and the first event with the given startTime is in higher
stream file sequence. The old logic will return no events, as the 
live stream reader expects one to keep polling for events (when
switching to new file, it will return 0 events and expect caller to call
read again. This works well for Flow). 

The new logic is simply to honor that behavior by keep
polling as long as there is events being read/skipped.

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java [hunk 1]
```diff
@@ -296,49 +316,65 @@ public P getPosition() {
...
+    private final long startTime;
+    private final long endTime;
+    private long hint;
+    private boolean active;
+
+    /**
+     * Creates a {@link TimeRangeReadFilter} with the specific time range.
+     *
+     * @param startTime start timestamp for event to be accepted (inclusive)
+     * @param endTime   end timestamp fo event to be accepted (exclusive)
```
Comment (ghelmling) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java]: "fo" -> "for"

Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java]: Good catch. Fixed.

**Comment by ghelmling** [comment]
one typo, otherwise lgtm.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java [hunk 2]
```diff
@@ -296,49 +316,65 @@ public P getPosition() {
...
+    private final long startTime;
+    private final long endTime;
+    private long hint;
+    private boolean active;
+
+    /**
+     * Creates a {@link TimeRangeReadFilter} with the specific time range.
+     *
+     * @param startTime start timestamp for event to be accepted (inclusive)
+     * @param endTime   end timestamp for event to be accepted (exclusive)
```
Comment (awholegunch) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java]: Remove extra spaces: ``@param endTime end timestamp for event to be accepted (exclusive)``

**Pull request merged**